fuse_core/include/fuse_core/throttled_callback.h

```diff
@@ -190,7 +176,35 @@ class ThrottledCallback
     callback(std::forward<Args>(args)...);
   }
 
-private:
+protected:
+  template <class... Args>
+  void callbackImpl(const ros::Time time, Args&&... args)
+  {
+    bool keep = false;
+    if (throttle_period_.isZero()) {
+      keep = true;
+    } else if (last_called_time_.isZero() || time > last_called_time_ + throttle_period_)
+    {
+      if (!last_called_time_.isZero()) {
+        keep = true;
+      }
+
+      // Use the nearest phase-aligned multiple of the throttle period as our reference time to improve determinism
+      const double time_s = time.toSec();
+      const double last_called_time_s =  time_s - std::fmod(time_s, throttle_period_.toSec());
+      last_called_time_.fromSec(last_called_time_s);
+    }
+
+    if (keep) {
+      if (keep_callback_) {
+        keep_callback_(std::forward<Args>(args)...);
+      }
+    } else if (drop_callback_)
```

Comment:
We use this style instead - the code LGTM
```suggestion
    if (throttle_period_.isZero())
    {
      keep = true;
    }
    else if (last_called_time_.isZero() || time > last_called_time_ + throttle_period_)
    {
      if (!last_called_time_.isZero())
      {
        keep = true;
      }

      // Use the nearest phase-aligned multiple of the throttle period as our reference time to improve determinism
      const double time_s = time.toSec();
      const double last_called_time_s =  time_s - std::fmod(time_s, throttle_period_.toSec());
      last_called_time_.fromSec(last_called_time_s);
    }

    if (keep)
    {
      if (keep_callback_)
      {
        keep_callback_(std::forward<Args>(args)...);
      }
    }
    else if (drop_callback_)
```

Reply:
I keep forgetting this :-). Having a hard time without my auto-formatter.